test/functional/sqlsrv/srv_231_string_truncation_text.phpt

```diff
@@ -0,0 +1,133 @@
+--TEST--
+GitHub issue #231 - String truncation when binding text/ntext/image to check error messages
+--ENV--
+PHPT_EXEC=true
+--SKIPIF--
+<?php require('skipif_versions_old.inc'); ?>
+--FILE--
+<?php
+
+sqlsrv_configure('WarningsReturnAsErrors', 1);
+
+require_once("MsCommon.inc");
+
+// connect
+$conn = AE\connect();
+if (!$conn) {
+    fatalError("Connection could not be established.\n");
+}
+
+$tableName = 'testLOBTypes_GH231';
+$columnNames = array("c1", "c2");
+
+for ($k = 1; $k <= 3; $k++) {
+    $sqlType = sqlType($k);
+    $columns = array(new AE\ColumnMeta('int', $columnNames[0]),
+                     new AE\ColumnMeta($sqlType, $columnNames[1]));
+    AE\createTable($conn, $tableName, $columns);
+
+    $sql = "INSERT INTO [$tableName] ($columnNames[0], $columnNames[1]) VALUES (?, ?)";
+    $data = getData($k);
+
+    $phpType = SQLSRV_PHPTYPE_STRING(SQLSRV_ENC_CHAR);
+    $sqlsrvSQLType = sqlsrvSqlType($k, strlen($data));
+    
+    $params = array($k, array($data, SQLSRV_PARAM_IN, $phpType, $sqlsrvSQLType));
+    $stmt = sqlsrv_prepare($conn, $sql, $params);
+    sqlsrv_execute($stmt);
+    sqlsrv_free_stmt($stmt);
+
+    execProc($conn, $tableName, $columnNames, $k, $data, $sqlType);
+
+    dropTable($conn, $tableName);
+}
+
+sqlsrv_close($conn);
+
+
+function execProc($conn, $tableName, $columnNames, $k, $data, $sqlType)
+{
+    $spArgs = "@p1 int, @p2 $sqlType OUTPUT";
+    $spCode = "SET @p2 = ( SELECT c2 FROM $tableName WHERE c1 = @p1 )";
+    $procName = "testBindOutSp";
+    
+    $stmt1 = sqlsrv_query($conn, "CREATE PROC [$procName] ($spArgs) AS BEGIN $spCode END");
+    sqlsrv_free_stmt($stmt1);
+
+    echo "\nData Type: ".$sqlType." binding as \n";
+
+    $direction = SQLSRV_PARAM_OUT;
+    echo "Output parameter: ";
+    invokeProc($conn, $procName, $k, $direction, $data);
+
+    $direction = SQLSRV_PARAM_INOUT;
+    echo "InOut parameter: ";
+    invokeProc($conn, $procName, $k, $direction, $data);
+
+    dropProc($conn, $procName);
+}
+
+function invokeProc($conn, $procName, $k, $direction, $data)
+{
+    $sqlsrvSQLType = sqlsrvSqlType($k, strlen($data));
+    $callArgs = "?, ?";
+
+    // Data to initialize $callResult variable
+    $initData = "ShortString";
+    $callResult = $initData;
+
+    // Make sure not to specify the PHP type
```

Comment:
Why must you not specify the php type here?

Reply:
It's a copy of another test for issue #231. I can remove the comment if you like. It doesn't matter for this particular test

Reply:
Then may as well remove it...